plugins/catalog-backend-module-github/src/providers/GithubMultiOrgEntityProvider.ts

```diff
@@ -118,14 +144,26 @@ export interface GithubMultiOrgEntityProviderOptions {
    * By default groups will be namespaced according to their GitHub org.
    */
   teamTransformer?: TeamTransformer;
+
+  /**
+   * An EventBroker to subscribe this provider to GitHub events to trigger delta mutations
+   */
+  eventBroker?: EventBroker;
 }
 
+type CreateDeltaOperation = (entities: Entity[]) => {
+  added: DeferredEntity[];
+  removed: DeferredEntity[];
+};
+
 /**
  * Ingests org data (users and groups) from GitHub.
  *
  * @public
  */
-export class GithubMultiOrgEntityProvider implements EntityProvider {
+export class GithubMultiOrgEntityProvider
+  implements EntityProvider, EventSubscriber
```

Comment:
Let's skip the addition of the `EventSubscriber` imo, it's the direction we're headed with the events API and avoids leaking the methods to the public interface. I think we should be able to just remove this and mark the methods private, or forward. Either way the following might be nice:

```ts
eventBroker.subscribe({
  supportsEventsTopics: () => [...],
  onEvent: provider.onEvent,
})
```

Reply:
hmm makes sense - is there an RFC I can stay up to date with the new design changes?

Reply:
Not right now, had those discussions in PRs. Depending on how much time he gets @pjungermann may or may not be working on an event system evolution RFC 😁, had a quite lengthy discussion around it at KubeCon EU